.github/workflows/deploy.yaml

```diff
@@ -38,7 +41,10 @@ jobs:
       - name: Terraform Init
         id: init
         working-directory: ./terraform
-        run: terraform init
+        run: |
+          STATE_PATH=${{ env.ENVIRONMENT == 'production' && 'prod' || env.ENVIRONMENT }}
+          terraform init \
+            -backend-config="key=terraform/state/${STATE_PATH}/terraform.tfstate"
```

Comment:
do we need `-backend-config` if we're using terraform workspace to select the appropriate key in `backend.tf`?